cognite/client/data_classes/units.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+
+import json
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from typing_extensions import Self
+
+from cognite.client.data_classes._base import CogniteResource, CogniteResourceList
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+@dataclass
+class UnitConversion:
+    """
+    The conversion between a unit and its base unit.
+
+    Args:
+        multiplier (float): The multiplier to convert from the unit to the base unit.
+        offset (float): The offset to convert from the unit to the base unit.
+    """
+
+    multiplier: float
+    offset: float
+
+    @classmethod
+    def load(cls, data: dict[str, Any]) -> Self:
+        return cls(
+            multiplier=data["multiplier"],
+            offset=data["offset"],
+        )
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        return {
+            "multiplier": self.multiplier,
+            "offset": self.offset,
+        }
+
+
+class UnitID(CogniteResource):
+    """
+    Unit Identifier
+
+    Args:
+        unit_external_id (str): External ID of the unit.
+        name (str): Name of the unit.
+    """
+
+    def __init__(self, unit_external_id: str, name: str) -> None:
+        self.unit_external_id = unit_external_id
+        self.name = name
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any] | str, cognite_client: CogniteClient | None = None) -> Self:
+        loaded = json.loads(resource) if isinstance(resource, str) else resource
+        return cls(
+            unit_external_id=loaded["unitExternalId"],
+            name=loaded["name"],
+        )
+
+
+class Unit(CogniteResource):
+    """
+    This class represents a Unit in CDF.
+
+    Args:
+        external_id (str): A unique identifier of the unit.
+        name (str): The name of the unit, e.g. °C for Celsius.
+        long_name (str): A more descriptive name of the unit, e.g., degrees Celsius.
+        alias_names (list[str]): List of alias names for the unit, e.g., Degree C, degC, °C, and so on.
+        quantity (str): The quantity of the unit, e.g., temperature.
+        conversion (UnitConversion): The conversion between the unit and its base unit. For example, the base unit for
+            temperature is Kelvin, and the conversion from Celsius to Kelvin is multiplier = 1, offset = 273.15.
+        source (str | None): The source of the unit, e.g., qudt.org
+        source_reference (str | None): The reference to the source of the unit, e.g., http://qudt.org/vocab/unit/DEG_C
+    """
+
+    def __init__(
```

Comment:
We'll have to add the `symbol` field here as with the update schema here: https://github.com/cognitedata/units-catalog/tree/main

Reply:
Done.